src/modules/admin/components/Permissions/ColonyPermissionEditDialog.tsx

```diff
@@ -0,0 +1,300 @@
+      displayName: filterValue,
+    },
+  };
+
+  return [customValue].concat(filtered);
+};
+
+const validationSchema = yup.object({
+  user: yup.object().required(),
+  domainId: yup.number(),
+  colonyAddress: yup.string().required(),
+  roles: yup.array().of(yup.string().required()),
+});
+
+const ColonyPermissionEditDialog = ({
+  domain,
+  /* selectedUser, */
+  colonyAddress,
+  cancel,
+  close,
+}: Props) => {
+  // Prepare userData for SingleUserPicker
+  const userAddressesInStore = useSelector(allUsersAddressesSelector);
+
+  const userData = useDataMapFetcher<UserType>(
+    usersByAddressFetcher,
+    Array.from(userAddressesInStore),
+  );
+
+  const users = useMemo(
+    () =>
+      userData
+        .filter(({ data }) => !!data)
+        .map(({ data, key }) => ({
+          id: key,
+          ...data,
+        })),
+    [userData],
+  );
+
+  const availableRoles = [
+    'Root',
+    'Administration',
+    'Architecture',
+    'Funding',
+    'Arbitration',
+  ];
+
+  /* Currently arbitration should appear in the list of roles
+    but can not be set yet
+   */
+  const checkIfCanBeSet = role => {
+    if (role === 'Arbitration' || role === 'Root') {
+      return false;
+    }
+    return true;
+  };
+
+  const transform = useCallback(
+    pipe(
+      withKey(colonyAddress),
+      mapPayload(p => ({
+        userAddress: p.user.profile.walletAddress,
+        domainId: p.domainId,
+        roles: p.roles.reduce((accumulator, role) => {
+          if (checkIfCanBeSet(role)) {
+            accumulator[role.toUpperCase()] = true;
+            return accumulator;
+          }
+          return accumulator;
+        }, {}),
+      })),
+      mergePayload({ colonyAddress }),
+    ),
+    [colonyAddress],
+  );
+
+  // When updating the selected user fetch that user's Roles
+  const [currentUser, setCurrentUser] = useState(null);
+
+  const updateCurrentUser = useCallback(({ profile: { walletAddress } }) => {
+    setCurrentUser(walletAddress);
+  }, []);
+
+  // Get current user roles to populate the checkboxes
+  const useCurrentUserForRoles = (): Array<string> => {
+    const { data } = useUserDomainRoles(
+      colonyAddress,
+      domain.id,
+      // selectedUser ||
+      currentUser,
+    );
+    if (data) {
+      return Object.keys(data).reduce((accumulator, role) => {
+        if (data[role]) {
+          // Role is capitalised in the ddb and needs to be readable
+          const roleLow = role.toLowerCase();
+          const readableRole =
+            roleLow.charAt(0).toUpperCase() + roleLow.slice(1);
+          accumulator.push(readableRole);
+          return accumulator;
+        }
+        return accumulator;
+      }, []);
+    }
+    return undefined;
+  };
+  const selectedRoles = useCurrentUserForRoles();
+
+  return (
+    <Dialog cancel={cancel}>
+      <ActionForm
+        initialValues={{
+          domainId: domain.id,
+          colonyAddress,
+          roles: selectedRoles,
```

Comment:
@sprusr I just noticed that the checkboxes in the dialog are being initialised before the `useUserDomainRoles` hook even returns the user roles. Any idea how to work around that?

Reply:
Maybe `selectedRoles && <ActionForm`...? So that it doesn't render until there is a value there. Otherwise can use `setValues` which is in `FormikProps` (available line 236 here).

Reply:
Will try, thank you for thinking about it!

Reply:
I found out what was missing, a`enableReinitialize` property on the ActionForm.